exercises/concept/lasagna-master/.meta/exemplar.js

```diff
@@ -0,0 +1,89 @@
+/// <reference path="./global.d.ts" />
+// @ts-check
+
+/**
+ * Determines whether the lasagna is done based on the
+ * remaining time on the timer.
+ *
+ * @param {number} time time left on the timer
+ * @returns {string} cooking status
+ */
+export function cookingStatus(time) {
+  if (time === undefined) {
+    return 'You forgot to set the timer.';
+  }
+
+  if (time === 0) {
+    return 'Lasagna is done.';
+  }
+
+  return 'Not done, please wait.';
+}
+
+/**
+ * Estimates the preparation time based on the number of layers.
+ *
+ * @param {string[]} layers
+ * @param {number} avgPrepTime
+ * @returns {number} total preparation time
+ */
+export function preparationTime(layers, avgPrepTime = 2) {
+  return layers.length * avgPrepTime;
+}
+
+/**
+ * Calculates how many noodles and much sauce are needed for the
+ * given layers.
+ *
+ * @param {string[]} layers
+ * @returns {Quantities} quantities needed for the given layers
+ */
+export function quantities(layers) {
```

Comment:
I would totally use reduce 👀

Reply:
and that's totally fine, the important thing here is just returning the object, how the student does the calculation does not matter

in the concept tree "reduce" is still like 2 layers away so I usually write the exemplar file only with what is covered so far